### ISSUE TITLE
Why do I get a loss over 1, when using the code from chapter 10 for the Fashion Mnist data set? 

### ISSUE THREAD

**Issue by anoldmaninthesea** [ageron/handson-ml2#132]
If I run the following code taken from chapter 10, I will get a loss greater than 1 when I evaluate the sequential neural network in the test set.

```
import tensorflow as tf
from tensorflow import keras
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

fashion_mnist=keras.datasets.fashion_mnist

(X_train_full,y_train_full),(X_test,y_test)=fashion_mnist.load_data()

X_valid , X_train = X_train_full[:5000]/255.0 , X_train_full[5000:]/255.0
y_valid , y_train = y_train_full[:5000] , y_train_full[5000:]

model=keras.models.Sequential()
model.add(keras.layers.Flatten(input_shape=[28,28]))
model.add(keras.layers.Dense(300,activation="relu"))
model.add(keras.layers.Dense(100,activation="relu"))
model.add(keras.layers.Dense(10,activation="softmax"))

model.compile(loss="sparse_categorical_crossentropy",optimizer="sgd",metrics=["accuracy"])
history=model.fit(X_train,y_train,epochs=30,validation_data=(X_valid,y_valid))

model.evaluate(X_test,y_test)
```

I get 

```
Train on 55000 samples, validate on 5000 samples
Epoch 1/30
55000/55000 [==============================] - 9s 156us/sample - loss: 0.7011 - accuracy: 0.7733 - val_loss: 0.5559 - val_accuracy: 0.8086
Epoch 2/30
55000/55000 [==============================] - 8s 136us/sample - loss: 0.4859 - accuracy: 0.8311 - val_loss: 0.4537 - val_accuracy: 0.8492
Epoch 3/30
55000/55000 [==============================] - 7s 133us/sample - loss: 0.4418 - accuracy: 0.8458 - val_loss: 0.4401 - val_accuracy: 0.8442
Epoch 4/30
55000/55000 [==============================] - 7s 135us/sample - loss: 0.4155 - accuracy: 0.8541 - val_loss: 0.4172 - val_accuracy: 0.8538
Epoch 5/30
55000/55000 [==============================] - 7s 133us/sample - loss: 0.3959 - accuracy: 0.8594 - val_loss: 0.3843 - val_accuracy: 0.8680
Epoch 6/30
55000/55000 [==============================] - 7s 134us/sample - loss: 0.3795 - accuracy: 0.8660 - val_loss: 0.3792 - val_accuracy: 0.8708
Epoch 7/30
55000/55000 [==============================] - 7s 133us/sample - loss: 0.3654 - accuracy: 0.8708 - val_loss: 0.3613 - val_accuracy: 0.8750
Epoch 8/30
55000/55000 [==============================] - 7s 134us/sample - loss: 0.3534 - accuracy: 0.8745 - val_loss: 0.3579 - val_accuracy: 0.8760
Epoch 9/30
55000/55000 [==============================] - 7s 134us/sample - loss: 0.3432 - accuracy: 0.8779 - val_loss: 0.3489 - val_accuracy: 0.8774
Epoch 10/30
55000/55000 [==============================] - 8s 143us/sample - loss: 0.3337 - accuracy: 0.8815 - val_loss: 0.3408 - val_accuracy: 0.8800
Epoch 11/30
55000/55000 [==============================] - 8s 139us/sample - loss: 0.3256 - accuracy: 0.8821 - val_loss: 0.3388 - val_accuracy: 0.8806
Epoch 12/30
55000/55000 [==============================] - 7s 132us/sample - loss: 0.3165 - accuracy: 0.8866 - val_loss: 0.3358 - val_accuracy: 0.8810
Epoch 13/30
55000/55000 [==============================] - 7s 133us/sample - loss: 0.3091 - accuracy: 0.8891 - val_loss: 0.3257 - val_accuracy: 0.8842
Epoch 14/30
55000/55000 [==============================] - 7s 131us/sample - loss: 0.3016 - accuracy: 0.8916 - val_loss: 0.3263 - val_accuracy: 0.8850
Epoch 15/30
55000/55000 [==============================] - 7s 132us/sample - loss: 0.2948 - accuracy: 0.8934 - val_loss: 0.3447 - val_accuracy: 0.8760
Epoch 16/30
55000/55000 [==============================] - 7s 130us/sample - loss: 0.2890 - accuracy: 0.8960 - val_loss: 0.3288 - val_accuracy: 0.8818
Epoch 17/30
55000/55000 [==============================] - 7s 132us/sample - loss: 0.2831 - accuracy: 0.8981 - val_loss: 0.3140 - val_accuracy: 0.8886
Epoch 18/30
55000/55000 [==============================] - 7s 129us/sample - loss: 0.2774 - accuracy: 0.8999 - val_loss: 0.3281 - val_accuracy: 0.8812
Epoch 19/30
55000/55000 [==============================] - 7s 134us/sample - loss: 0.2722 - accuracy: 0.9013 - val_loss: 0.3166 - val_accuracy: 0.8880
Epoch 20/30
55000/55000 [==============================] - 7s 130us/sample - loss: 0.2666 - accuracy: 0.9024 - val_loss: 0.3240 - val_accuracy: 0.8810
Epoch 21/30
55000/55000 [==============================] - 7s 132us/sample - loss: 0.2620 - accuracy: 0.9049 - val_loss: 0.3006 - val_accuracy: 0.8910
Epoch 22/30
55000/55000 [==============================] - 7s 131us/sample - loss: 0.2569 - accuracy: 0.9061 - val_loss: 0.3133 - val_accuracy: 0.8876
Epoch 23/30
55000/55000 [==============================] - 7s 130us/sample - loss: 0.2515 - accuracy: 0.9087 - val_loss: 0.3096 - val_accuracy: 0.8892
Epoch 24/30
55000/55000 [==============================] - 7s 131us/sample - loss: 0.2474 - accuracy: 0.9104 - val_loss: 0.3018 - val_accuracy: 0.8922
Epoch 25/30
55000/55000 [==============================] - 7s 130us/sample - loss: 0.2427 - accuracy: 0.9120 - val_loss: 0.3249 - val_accuracy: 0.8834
Epoch 26/30
55000/55000 [==============================] - 7s 130us/sample - loss: 0.2388 - accuracy: 0.9134 - val_loss: 0.3032 - val_accuracy: 0.8894
Epoch 27/30
55000/55000 [==============================] - 7s 129us/sample - loss: 0.2348 - accuracy: 0.9151 - val_loss: 0.2926 - val_accuracy: 0.8956
Epoch 28/30
55000/55000 [==============================] - 7s 130us/sample - loss: 0.2305 - accuracy: 0.9168 - val_loss: 0.2954 - val_accuracy: 0.8954
Epoch 29/30
55000/55000 [==============================] - 7s 128us/sample - loss: 0.2270 - accuracy: 0.9181 - val_loss: 0.2963 - val_accuracy: 0.8940
Epoch 30/30
55000/55000 [==============================] - 7s 130us/sample - loss: 0.2233 - accuracy: 0.9199 - val_loss: 0.2949 - val_accuracy: 0.8950
10000/10000 [==============================] - 1s 109us/sample - loss: 71.0087 - accuracy: 0.8315
[71.00866251907348, 0.8315]
```

**Comment by anoldmaninthesea** [comment]
You forgot to divide /255.0  in the X_test
